ouroboros-network/src/Codec/SerialiseTerm.hs

```diff
@@ -0,0 +1,12 @@
+module Codec.SerialiseTerm where
+
+import           Data.Text (Text)
+import           Codec.CBOR.Term as CBOR
+
+-- |
+-- Serialisation and de-serialisation for CBOR-in-CBOR encoding using
+-- @'CBOR.Term'@.
+--
+class SerialiseTerm a where
```

Comment:
I request using a record type for this instead.

```Haskell
data CodecCBORTerm fail a = CodecCBORTerm
  { encodeTerm :: a -> CBOR.Term
  , decodeTerm :: CBOR.Term -> Either fail a
  }
```

 If there are applications in which having this typeclass is significantly better, it can be supplied by `class SerialiseTerm a where serialiseTerm :: CodecCBORTerm Text a`

Reply:
That's fine.

Reply:
Fixed